girder/cli/build.py

```diff
@@ -22,27 +22,33 @@
 
 
 @click.command(name='build', help='Build web client static assets.')
-@click.option('--dev/--no-dev', default=False,
-              help='Build girder client for development.')
+@click.option('--dev', default=False, is_flag=True, help='Sets the server mode to development')
+@click.option('--mode', type=click.Choice([PRODUCTION_MODE, DEVELOPMENT_MODE]),
+              default=PRODUCTION_MODE, show_default=True, help='Specify the server mode')
 @click.option('--watch', default=False, is_flag=True,
-              help='Build girder library bundle in watch mode (implies --dev --no-reinstall).')
+              help='Build girder library bundle in '
+              'watch mode (implies mode=development --no-reinstall).')
 @click.option('--watch-plugin',
-              help='Build a girder plugin bundle in watch mode (implies --dev --no-reinstall).')
+              help='Build a girder plugin bundle in '
+              'watch mode (implies mode=development --no-reinstall).')
 @click.option('--npm', default=os.getenv('NPM_EXE', 'npm'),
               help='Full path to the npm executable to use.')
 @click.option('--reinstall/--no-reinstall', default=True,
               help='Force regenerate node_modules.')
-def main(dev, watch, watch_plugin, npm, reinstall):
+def main(dev, mode, watch, watch_plugin, npm, reinstall):
     if shutil.which(npm) is None:
         raise click.UsageError(
             'No npm executable was detected.  Please ensure the npm executable is in your '
             'path, use the --npm flag, or set the "NPM_EXE" environment variable.'
         )
 
+    if (dev):
+        mode = DEVELOPMENT_MODE
```

Comment:
If both `--dev` and `--mode` are set, should we emit an error / warning?

Reply:
Same as above